cardano-node/src/Cardano/CLI/Ops.hs

```diff
@@ -221,3 +266,139 @@ withRealPBFT gHash genFile nMagic sigThresh delCertFp sKeyFp update ptcl action
+  -> SocketPath
+  -> IO ()
+createNodeConnection proxy ptcl socketPath = do
+    addr <- localSocketAddrInfo socketPath
+    let ProtocolInfo{pInfoConfig} = protocolInfo ptcl
+    connectTo
+      (NetworkConnectTracers nullTracer nullTracer)
+      (localInitiatorNetworkApplication proxy pInfoConfig)
+      Nothing
+      addr
+    `catch` handleMuxError
+
+handleMuxError :: MuxError -> IO ()
+handleMuxError err = print err
+
+localInitiatorNetworkApplication
+  :: forall blk m peer.
+     ( RunNode blk
+     , Condense (HeaderHash blk)
+     , IOLike m
+     , MonadIO m
+     , MonadTimer m
+     )
+  => Proxy blk
+  -> Consensus.NodeConfig (BlockProtocol blk)
+  -> Versions NodeToClientVersion DictVersion
+              (OuroborosApplication 'InitiatorApp peer NodeToClientProtocols
+                                    m LB.ByteString () Void)
+localInitiatorNetworkApplication proxy pInfoConfig =
+    simpleSingletonVersions
+      NodeToClientV_1
+      (NodeToClientVersionData { networkMagic = nodeNetworkMagic proxy pInfoConfig })
+      (DictVersion nodeToClientCodecCBORTerm)
+
+  $ OuroborosInitiatorApplication $ \_peer ptcl -> case ptcl of
+      LocalTxSubmissionPtcl -> \channel ->
+        runPeer
+          nullTracer
+          localTxSubmissionCodec
+          channel
+          (localTxSubmissionClientPeer localTxSubmissionClientNull)
+
+      ChainSyncWithBlocksPtcl -> \channel ->
+        runPeer
+          nullTracer
+          localChainSyncCodec
+          channel
+          (chainSyncClientPeer chainSyncClient)
+ where
+  localChainSyncCodec :: Codec (ChainSync (Serialised blk) (Tip blk)) DeserialiseFailure m LB.ByteString
+  localChainSyncCodec = pcLocalChainSyncCodec . protocolCodecs pInfoConfig $ mostRecentNetworkProtocolVersion proxy
+
+  localTxSubmissionCodec :: Codec (LocalTxSubmission (GenTx blk) (ApplyTxErr blk)) DeserialiseFailure m LB.ByteString
+  localTxSubmissionCodec = pcLocalTxSubmissionCodec . protocolCodecs pInfoConfig $ mostRecentNetworkProtocolVersion proxy
+
+
+chainSyncClient
+  :: forall blk m . (Condense (HeaderHash blk), MonadIO m)
+  => ChainSyncClient (Serialised blk) (Tip blk) m ()
+chainSyncClient = ChainSyncClient $ pure $
+  SendMsgRequestNext
+    clientStNext
+    (pure $ ClientStNext
+              { recvMsgRollForward = \_ _ -> ChainSyncClient $ pure clientStIdle
+              , recvMsgRollBackward = \_ _ -> ChainSyncClient $ pure clientStIdle
+              }
+    )
+ where
+  clientStIdle :: ClientStIdle (Serialised blk) (Tip blk) m ()
+  clientStIdle =
+    SendMsgRequestNext clientStNext (pure clientStNext)
+
+  clientStNext :: ClientStNext (Serialised blk) (Tip blk) m ()
+  clientStNext = ClientStNext
+    { recvMsgRollForward = \_blk tip -> ChainSyncClient $ do
+        traceWith stdoutTracer . toS $ getTipOutput tip
+        pure $ SendMsgDone ()
+    , recvMsgRollBackward = \_point tip -> ChainSyncClient $ do
+        traceWith stdoutTracer . toS $ getTipOutput tip
```

Comment:
Add a `TODO` for these two traces: we should be able to simply return the tip as the result with `SendMsgDone` and collect this as the result of the overall protocol. While currently we can have protocols return things, the current `OuroborosApplication` stuff gets in the way of returning an overall result, but that's being worked on, and this can be improved when that's ready.